src/Adapter/Tax/TaxComputer.php

```diff
@@ -100,15 +101,19 @@ public function computePriceWithoutTaxes(
      */
     public function getTaxRate(TaxRulesGroupId $taxRulesGroupId, CountryId $countryId): DecimalNumber
     {
-        $taxRulesGroup = $this->taxRulesGroupRepository->getTaxRulesGroupDetails($taxRulesGroupId);
-        if (!empty($taxRulesGroup['rates'])) {
-            // Use the tax rate associated to context country, or the first one as fallback
-            $countryTaxRate = $taxRulesGroup['rates'][$countryId->getValue()] ?? reset($taxRulesGroup['rates']);
-        } else {
-            $countryTaxRate = 0;
+        $country = new \Country($countryId->getValue());
```

Comment:
Inject [CountryRepository](https://github.com/PrestaShop/PrestaShop/blob/8.0.x/src/Adapter/Country/Repository/CountryRepository.php) into constructor and use $this->countryRepository->get($countryId) instead. 
Though this method doesn't exist yet so you'll need to implement it. (its pretty easy, just check other example classes that extends AbstractObjectModelRepository)